src/library/scala/collection/mutable/HashMap.scala

```diff
@@ -161,7 +161,13 @@ class HashMap[K, V](initialCapacity: Int, loadFactor: Double)
           else table(indexedHash) = foundNode.next
           contentSize -= 1
 
-        case (None, Some(value)) => put0(key, value, false, hash, indexedHash)
+        case (None, Some(value)) =>
+          if(contentSize + 1 >= threshold) {
+            growTable(table.length * 2)
+            val newIndexedHash = index(hash)
+            put0(key, value, false, hash, newIndexedHash)
+          }
+          else put0(key, value, false, hash, indexedHash)
```

Comment:
It could be simplified to:
```      val newIndexedHash = 
            if(contentSize + 1 >= threshold) {
              growTable(table.length * 2)
              index(hash)
            } else indexedHash
            put0(key, value, false, hash, newIndexedHash)
          }
          put0(key, value, false, hash, indexedHash)
```

Reply:
Indeed!

Reply:
Uhm that I think I consider as being a little less readable...